keps/sig-node/4381-dra-semantic-parameters/README.md

```diff
@@ -0,0 +1,1477 @@
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `<package>`: `<date>` - `<test coverage>`
+
+##### Integration tests
+
+<!--
+Integration tests are contained in k8s.io/kubernetes/test/integration.
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+- <test>: <link to test coverage>
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Fully implemented
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- 3 examples of real-world usage
+- Allowing time for feedback
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+### Upgrade / Downgrade Strategy
+
+Because of the strongly-typed versioning of resource attributes and allocation
+results, the gRPC interface between kubelet and the DRA driver is tied to the
+version of the supported semantic models. A DRA driver has to implement all
+gRPC interfaces that might be used by older releases of kubelet. The same
+applies when upgrading kubelet while the DRA driver remains at an older
+version.
+
+### Version Skew Strategy
+
+Ideally, the latest release of a DRA driver should be used and it should
+support a wide range of semantic type versions. Then problems due to version
+skew are less likely to occur.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate
+  - Feature gate name: DynamicResourceAllocation
+  - Components depending on the feature gate:
+    - kube-apiserver
+    - kubelet
+    - kube-scheduler
+    - kube-controller-manager
+
+###### Does enabling the feature change any default behavior?
+
```

Comment:
We need to answer these - they likely are the same answers for base DRA.

Reply:
Ack. I went through those and they indeed seem applicable, so I copied them over.

I stopped at those entries which are required "when targeting beta to a release" because we are not doing that now.